src/main/java/io/cdap/plugin/gcp/bigquery/util/BigQueryUtil.java

```diff
@@ -183,6 +183,17 @@ public static void validateFieldSchemaMatches(Field bqField, Schema.Field field,
                         field.getName(), logicalType, bqField.getName(), dataset, table,
                         bqField.getType(), bqField.getType()));
       }
+
+      // BigQuery requires schema precision to be 38 and scale 9
+      if (logicalType == Schema.LogicalType.DECIMAL) {
+        if (fieldSchema.getPrecision() != 38 || fieldSchema.getScale() != 9) {
```

Comment:
so it doesn't allow lower precision and lower scale? If it's lower the plugin can adjust it up can't it?

Reply:
To start out with, I think the plugin should support whatever the outputformat supports.

If it has to be exactly 38 and 9, then let's keep this check but change the error message to: 
Numeric field '%s' has invalid precision '%d' and scale '%d'. Precision must be 38 and scale must be 9.

If it allows smaller values, then this should be changed to > checks and the error message should be:
Numeric field '%s' has invalid precision '%d' and scale '%d'. Precision must be at most 38 scale must be at most 9.